### PR TITLE
Add support for usermod palettes

### PR DESCRIPTION
**Pull request by mik-laj**
# Proposed Changes

> (Describe the changes and rationale behind them)

This PR adds support for synthesizing “usermod palettes” into the Device.palettes mapping based on firmware-reported counts/names. It follows custom palettes flow.

## Related Issues

> ([GitHub link][autolink-references] to related issues or pull requests)

[autolink-references]: https://help.github.com/articles/autolinked-references-and-urls/

## Summary by CodeRabbit

* **New Features**
  * Added support for usermod (user-created) palettes with automatic fallback naming on compatible firmware versions.
  * Extended palette management to properly synthesize and display usermod palette entries alongside built-in and custom palettes.

* **Tests**
  * Added comprehensive test coverage for usermod palette behavior, including edge cases and version compatibility.

[Review Change Stack](https://app.coderabbit.ai/change-stack/frenck/python-wled/pull/2058)

### CODE REVIEW

File: tests/test_models.py [hunk 1]
```diff
@@ -685,16 +685,100 @@ def test_device_update_from_dict_effects() -> None:
...
-    """Test update_from_dict updates palettes."""
+@pytest.mark.parametrize(
+    ("version", "expected_palette_ids"),
+    [
+        ("0.14.0", [255, 254]),
+        ("15.0.0", [255, 254]),
+        ("16.0.0", [200, 199]),
+        ("16.5.0", [200, 199]),
+        ("17.0.0", [200, 199]),
+    ],
```
[the remark this answers is not in the document]
Comment (mik-laj) [reply tests/test_models.py:712]: WLED changed versioning schema. It is correct value.

File: src/wled/const.py [hunk 2]
```diff
@@ -8,6 +8,8 @@
 
 MIN_REQUIRED_VERSION = AwesomeVersion("0.14.0")
 
+CUSTOM_PALETTE_ID_CHANGE_VERSION = AwesomeVersion("16.0.0")
```
[the remark this answers is not in the document]
Comment (mik-laj) [reply src/wled/const.py:11]: Not. It is not correct.

**Review by frenck — APPROVED** [review]
